File: app/src/main/java/com/babestudios/companyinfouk/ui/officers/OfficersActivityView.java
package com.babestudios.companyinfouk.ui.officers;

import com.babestudios.companyinfouk.data.model.officers.Officers;

import net.grandcentrix.thirtyinch.TiView;
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread;

interface OfficersActivityView extends TiView {

	@CallOnMainThread
	void showProgress();

	@CallOnMainThread
	void hideProgress();

	@CallOnMainThread
	void showError();

	@CallOnMainThread
	void showOfficers(final Officers officers);

	@CallOnMainThread
	String getCompanyNumber();

}

File: app/src/main/java/com/babestudios/companyinfouk/ui/officers/OfficersActivityView.kt
package com.babestudios.companyinfouk.ui.officers

import com.babestudios.companyinfouk.data.model.officers.Officers

import net.grandcentrix.thirtyinch.TiView
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread

interface OfficersActivityView : TiView {

	@get:CallOnMainThread
	val companyNumber: String

	@CallOnMainThread
	fun showProgress()

	@CallOnMainThread
	fun hideProgress()

	@CallOnMainThread
	fun showError()

	@CallOnMainThread
	fun showOfficers(officers: Officers)

}